src/main/java/wolox/training/models/User.java

```diff
@@ -53,13 +53,18 @@ public void setBooks(Collection<Book> books) {
         this.books = books;
     }
 
-    public Collection<Book> addBook(Book newBook) {
-        this.books.add(newBook);
-        return this.books;
+    public boolean addBook(Book newBook) {
+        if(this.books.stream().anyMatch(book -> book.getId() == newBook.getId())){
```

Comment:
All of this can get replaced for a simple "return this.books.add(newBook);"
That is, of course, assuming you implement the "equals" and "hashcode" in the Book model correctly.